### ISSUE TITLE
`Cannot set property 'textContent' of null` within JSX expression

### ISSUE THREAD

**Issue by bluenote10** [ryansolid/solid#44]
I've been scratching my head about some rendering bugs I'm facing. When trying to come up with a minimal reproducing example of the actual issue (in a more complex scenario) I came up with something that has a different error/traceback, but might be related. I'm again not sure if I'm doing something wrong, or something is broken.

Example / [Codepen](https://codesandbox.io/s/debugrenderjsx-0z58q):
```ts
export function View(props: { title: string }) {
  return (
    <div>
      <h1>{(props.title)}</h1>
      <table>
        <tr>
          <td><b>Title</b></td>
          <td>
            {(props.title)}
          </td>
        </tr>
      </table>
    </div>
  )
}

function Debug() {
  return (
    <div>
      <View title="test"/>
    </div>
  )
}
```

This errors in the JSX expression:

```
index.js:176 Uncaught TypeError: Cannot set property 'textContent' of null
    at insertExpression (index.js:176)
    at Object (index.js:195)
    at execToplevelComputation (solid.js:294)
    at createEffect (solid.js:36)
    at insert (index.js:195)
    at app_renderer.tsx:66
    at View (app_renderer.tsx:66)
    at app_renderer.tsx:84
    at Debug (app_renderer.tsx:84)
    at ./src/app_renderer.tsx.Object (app_renderer.tsx:93)
```

Maybe clarifying the problem of this simpler example helps to narrow down the problem in the full example.

Most likely the issue is in the babel-jsx plugin? Do you actually prefer tickets like these in this repo or in the babel-jsx repo when it's not fully clear if it's on solid side or babel-jsx?

**Comment by ryansolid** [comment]
I see what's happening here although I'm not stoked about it. Basically the generated HTML is different than the template. Chrome and I imagine many browsers automatically add `<tbody>` elements when the innerHTML is set which messes with the parsers ability to walk the tree at compile time since it's not there.

It might be possible to to detect this in the template and always add tbody if it is missing, but I'm unsure of what other gotchas we are going to hit like this which browser is a lot more forgiving of differently formed html (fixing it as it sees fit) differing from the template actually provided.

**Comment by bluenote10** [comment]
:+1: thanks for the clarification! I wasn't aware of such behavior and thought using `tbody` is optional for tables without a header because it always worked fine in other frameworks. Even if I knew it, I would have expected to just render the malformed HTML.

I'm wondering if there are more of these cases of magically added nodes? I found at least one other: If you omit the `table` tag altogether, but have a `tbody` it also crashes.

**Comment by ryansolid** [comment]
Yeah it is going to be difficult to impossible for me to handle all the cases. I'm not sure how many I should chase after. Maybe I could do some detection of html structure at compile time and throw errors so at least there is no runtime issue. However, it would be difficult to guarantee jsdom etc would have the same output. I could also do a check at runtime on template creation. Maybe that's enough for now.

Basically there is code that looks like:
```js
const template = document.createElement('template');
template.innerHTML = templateString;
```
I could do a check at that point to ensure `template.innerHTML === templateString`. It puts the onus on the developer to make well formed JSX. Opinion?

**Comment by bluenote10** [comment]
Yes, a clear error message seems to be the best option, either at compile time or run time.

**Comment by ryansolid** [comment]
Fixed with a runtime check in v0.9.0.
